api_app/analyzers_manager/file_analyzers/elf_info.py

```diff
@@ -0,0 +1,36 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+import logging
+
+from elftools.common.exceptions import ELFError
+from elftools.elf.elffile import ELFFile
+from telfhash import telfhash
+
+from api_app.analyzers_manager.classes import FileAnalyzer
+
+logger = logging.getLogger(__name__)
+
+
+class ELFInfo(FileAnalyzer):
+    def run(self):
+        results = {}
+        try:
+            with open(self.filepath, "rb") as file:
+                elf = ELFFile(file)
+            if elf is None:
+                raise ELFError("Not an ELF file")
+            results["telf"] = telfhash(self.filepath)[0]
```

Comment:
how can you be sure that this list is populated? please handle this possible error

Reply:
![image](https://user-images.githubusercontent.com/58381523/177567250-05923e27-ca16-4904-8a4f-204ef43eef39.png)

Reply:
yeah but the point is that you have no control over that library so you never know how things can change and this would result in an unmanaged exception error.

Even if yeah it is not a particular problem, I would suggest to manually raise an explicit error in case that result is not a populated list.

This also helps in case we extend this analyzer more to add additional components.